### PR TITLE
store/tikv: add batch cleanup, update commit response

### PR DESCRIPTION
**Pull request by disksing**
After https://github.com/pingcap/kvproto/pull/57
@ngaut @morefreeze

### CODE REVIEW

File: store/tikv/txn_committer.go [hunk 1]
```diff
@@ -231,21 +229,34 @@ func (c *txnCommitter) commitSingleRegion(keys [][]byte) error {
...
+	}
+	resp, err := c.store.SendKVReq(req, region)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if regionErr := resp.GetRegionError(); regionErr != nil {
+		err = c.cleanupKeys(keys)
+		return errors.Trace(err)
+	}
+	rollbackResp := resp.GetCmdCleanupResp()
```
Comment (morefreeze) [inline store/tikv/txn_committer.go]: `resp.GetCmdBatchRollbackResp()` ?

File: store/tikv/txn_committer.go [hunk 2]
```diff
@@ -231,21 +229,34 @@ func (c *txnCommitter) commitSingleRegion(keys [][]byte) error {
...
-		// TODO: check response and retry
+	req := &pb.Request{
+		Type: pb.MessageType_CmdBatchRollback.Enum(),
+		CmdBatchRollbackReq: &pb.CmdBatchRollbackRequest{
+			Keys:         keys,
+			StartVersion: proto.Uint64(c.startTS),
+		},
+	}
+
+	region, err := c.store.getRegion(keys[0])
```
Comment (morefreeze) [inline store/tikv/txn_committer.go]: Check `len(keys) > 0` is better.

Comment (disksing) [inline store/tikv/txn_committer.go]: @morefreeze The algorithm guarantees it won't be an empty slice. By the way, I will refactor it soon and use an ID to get region.

**Comment by morefreeze** [comment]
LGTM

File: store/tikv/txn_committer.go [hunk 3]
```diff
@@ -231,21 +229,34 @@ func (c *txnCommitter) commitSingleRegion(keys [][]byte) error {
...
+	if err != nil {
+		return errors.Trace(err)
+	}
+	resp, err := c.store.SendKVReq(req, region)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if regionErr := resp.GetRegionError(); regionErr != nil {
```
Comment (shenli) [inline store/tikv/txn_committer.go]: Set limitation for retry?

**Comment by shenli** [comment]
Rest LGTM

File: store/tikv/txn_committer.go (hunk 3 above)
Comment (disksing) [inline store/tikv/txn_committer.go]: @shenli Already had a TODO (L153) for that.

**Comment by shenli** [comment]
LGTM

**Pull request merged**